### PR TITLE
Add spring-boot-starter-oauth2-resource-server to contentgrid-spring-data-rest tests

### PR DESCRIPTION
**Pull request by NielsCW**

### CODE REVIEW

File: contentgrid-spring-data-rest/build.gradle [hunk 1]
```diff
@@ -42,7 +42,10 @@ dependencies {
     testImplementation 'org.springframework.boot:spring-boot-starter-test'
     testImplementation 'org.springframework.boot:spring-boot-starter-web'
     testImplementation 'jakarta.transaction:jakarta.transaction-api'
+    testImplementation 'org.springframework.security:spring-security-test'
+    testImplementation 'org.springframework.boot:spring-boot-starter-oauth2-resource-server'
     testImplementation project(':contentgrid-spring-boot-autoconfigure')
+    testImplementation project(':contentgrid-spring-data-support')
```
Comment (vierbergenlars) [inline contentgrid-spring-data-rest/build.gradle:48]: It doesn't look like you're using any classes from this project in your tests. Do you need this dependency?

Comment (NielsCW) [reply contentgrid-spring-data-rest/build.gradle:48]: Not now in this PR, it is needed when adding audit metadata tests

File: contentgrid-spring-data-rest/src/test/java/com/contentgrid/spring/test/security/WithMockJwt.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import org.springframework.security.test.context.support.WithSecurityContext;
+import org.springframework.security.test.context.support.WithSecurityContextFactory;
+import org.springframework.util.StringUtils;
+
+@Target({ElementType.METHOD, ElementType.TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+@Inherited
+@Documented
+@WithSecurityContext(factory = WithMockJwt.WithMockJwtSecurityContextFactory.class)
+@Import({SkipCsrfConfiguration.class})
```
Comment (tgeens) [inline contentgrid-spring-data-rest/src/test/java/com/contentgrid/spring/test/security/WithMockJwt.java:29]: Now I'm wondering what happens with this import on the method-level ? :thinking:
That won't work, right ?

Comment (NielsCW) [reply contentgrid-spring-data-rest/src/test/java/com/contentgrid/spring/test/security/WithMockJwt.java:29]: Removing the `@WithMockJwt` on the class and placing it on the method testing POST -> test fails with http 403

Comment (vierbergenlars) [reply contentgrid-spring-data-rest/src/test/java/com/contentgrid/spring/test/security/WithMockJwt.java:29]: If it's not supported on methods, then `@Target` should not include `ElementType.METHOD`

Comment (tgeens) [reply contentgrid-spring-data-rest/src/test/java/com/contentgrid/spring/test/security/WithMockJwt.java:29]: but it is intended to be used on method level too ...

Comment (NielsCW) [reply contentgrid-spring-data-rest/src/test/java/com/contentgrid/spring/test/security/WithMockJwt.java:29]: Setting the `@WithMockJwt` on the method overrides the properties of the one on the class. It is when the `@WithMockJwt` is missing on the class that the `SkipCsrfConfiguration` does not get loaded.

Comment (tgeens) [reply contentgrid-spring-data-rest/src/test/java/com/contentgrid/spring/test/security/WithMockJwt.java:29]: In that case, you might need to use `@Import(SkipCsrfConfiguration.class)` on the test-class first. 

Ok, fine with me.

Comment (tgeens) [inline contentgrid-spring-data-rest/src/test/java/com/contentgrid/spring/test/security/WithMockJwt.java:29]:
```suggestion
@Import(SkipCsrfConfiguration.class)
```

File: contentgrid-spring-data-support/build.gradle [hunk 3]
```diff
@@ -24,7 +24,7 @@ dependencies {
     implementation 'jakarta.persistence:jakarta.persistence-api'
     implementation 'com.fasterxml.jackson.core:jackson-annotations'
     implementation 'org.springframework.data:spring-data-jpa'
-    compileOnly 'org.springframework.security:spring-security-oauth2-jose'
+    implementation 'org.springframework.security:spring-security-oauth2-jose'
```
Comment (tgeens) [inline contentgrid-spring-data-support/build.gradle:27]: why is this change required ?

**Review by tgeens — APPROVED** [review]
